x/stakeibc/keeper/unbonding_records_get_host_zone_unbondings_msgs_test.go

```diff
@@ -1,226 +1,813 @@
+
+	// Sanity checks:
+	//  - total stake matches
+	//  - total weights sum to 100
+	actualTotalStake := sdkmath.ZeroInt()
+	actualTotalWeights := uint64(0)
+	for _, validator := range validators {
+		actualTotalStake = actualTotalStake.Add(validator.Delegation)
+		actualTotalWeights += validator.Weight
+	}
+	s.Require().Equal(totalStake.Int64(), actualTotalStake.Int64(), "test setup failed - total stake does not match")
+	s.Require().Equal(totalWeight, int64(actualTotalWeights), "test setup failed - total weight does not match")
+
+	// Store the validators on the host zone
+	hostZone := types.HostZone{
+		ChainId:              HostChainId,
+		ConnectionId:         ibctesting.FirstConnectionID,
+		HostDenom:            Atom,
+		DelegationIcaAddress: "cosmos_DELEGATION",
+		Validators:           validators,
+	}
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx, hostZone)
+
+	// Store the total unbond amount across two epoch unbonding records
+	halfUnbondAmount := unbondAmount.Quo(sdkmath.NewInt(2))
+	for i := uint64(1); i <= 2; i++ {
+		s.App.RecordsKeeper.SetEpochUnbondingRecord(s.Ctx, recordstypes.EpochUnbondingRecord{
+			EpochNumber: i,
+			HostZoneUnbondings: []*recordstypes.HostZoneUnbonding{
+				{
+					HostZoneId:        HostChainId,
+					Status:            recordstypes.HostZoneUnbonding_UNBONDING_QUEUE,
+					NativeTokenAmount: halfUnbondAmount,
+				},
+			},
+		})
+	}
+
+	// Mock the epoch tracker to timeout 90% through the epoch
+	strideEpochTracker := types.EpochTracker{
+		EpochIdentifier:    epochstypes.DAY_EPOCH,
+		Duration:           10_000_000_000,                                                // 10 second epochs
+		NextEpochStartTime: uint64(s.Coordinator.CurrentTime.UnixNano() + 30_000_000_000), // dictates timeout
+	}
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx, strideEpochTracker)
+
+	// Get tx seq number before the ICA was submitted to check whether an ICA was submitted
+	startSequence, found := s.App.IBCKeeper.ChannelKeeper.GetNextSequenceSend(s.Ctx, delegationPortID, delegationChannelID)
+	s.Require().True(found, "sequence number not found before ica")
+
+	return UnbondingTestCase{
+		hostZone:                   hostZone,
+		totalUnbondAmount:          unbondAmount,
+		delegationChannelID:        delegationChannelID,
+		delegationPortID:           delegationPortID,
+		channelStartSequence:       startSequence,
+		expectedUnbondingRecordIds: []uint64{1, 2},
+	}
+}
+
+// Helper function to check that an undelegation ICA was submitted and that the callback data
+// holds the expected unbondings for each validator
+func (s *KeeperTestSuite) CheckUnbondingMessages(tc UnbondingTestCase, expectedUnbondings []ValidatorUnbonding) {
+	// Trigger unbonding
+	err := s.App.StakeibcKeeper.UnbondFromHostZone(s.Ctx, tc.hostZone)
+	s.Require().NoError(err, "no error expected when calling unbond from host")
+
+	// Check that sequence number incremented from a sent ICA
```

Comment:
Why use the sequence # to check whether an ICA was submitted, rather than checking some result from the ICA directly? Haven't seen this approach before.

Reply:
did you have anything specific in mind? the ICA isn't relayed here so there's not a lot that we can check

Reply:
I can't think of another way, this is creative!